### PR TITLE
Generator pq

### PR DESCRIPTION
**Pull request by dimbdr**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Introduce the PQ generator

### CODE REVIEW

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.powsybl.dynawaltz.dynamicmodels.staticref.generators.omegaref;
+
+import com.powsybl.dynawaltz.dynamicmodels.staticref.generators.GeneratorModel;
+
+public interface GeneratorConnectedToOmegaRefModel extends GeneratorModel {
+    default String getOmegaRefPuVarName() {
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:6]: We try to use default keyword only for retrocompatibility: if an interface evolves with a new method we can give a default implementation so that users don't need to add anything in their old implementation. Here that's not the problem, so that means that you need to add a, wait for it, `AbstractGeneratorConnectedToOmegaRefModel` class :face_exhaling:

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.powsybl.dynawaltz.dynamicmodels.staticref.generators.omegaref;
+
+import com.powsybl.dynawaltz.dynamicmodels.staticref.generators.GeneratorModel;
+
+public interface GeneratorConnectedToOmegaRefModel extends GeneratorModel {
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:5]: isn't there a more specific word for these kind of generators?

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.powsybl.dynawaltz.dynamicmodels.staticref.generators.omegaref;
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:1]: You can add license header and authorship. To ease up the license scanning done by the linux foundation, we've been recently asked to add a "SPDX identifier":

```
/**
 * Copyright (c) 2022, RTE (http://www.rte-france.com/)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 * SPDX-License-Identifier: MPL-2.0
 */
```

File: dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/dynamicmodels/GeneratorPQGroovyExtension.groovy [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (flo-dup) [inline dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/dynamicmodels/GeneratorPQGroovyExtension.groovy:2]: Could you fix the header (2022 + SPDX identifier) and add authorship?

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java (hunk 2 above)
Comment (flo-dup) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:5]: `AsynchronousGeneratorModel`?

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java (hunk 1 above)
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:6]: Ah damn, I thought I could avoid this abstract.
It's quite an issue. For each synchronous generator, we would have 6 inheritance levels. Sonar is not really happy about it.
I don't have any idea right now how to fix this.

Comment (geofjamg) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:6]: Sonar issues are not alway relevant. In many places in powsybl, we do not follow this 6 levels recommandation.
For some complex data model, it could be ok to be greater than 6

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java (hunk 2 above)
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:5]: I don't have the full picture.
@gautierbureau and I talked a bit about the names, still it's not really clear how to name correctly all those objects.

tl;dr : we'll not use `AsynchronousGeneratorModel`, sorry :cry: 


**Long answer, it's how I understand it:**
After we discussed, the name `SynchronousGenerator` is more about a state than a definition. A synchronous generator might be non-synchronous at some point in time.
But we do not concern ourselves too much about this situation, in the initial state of the model context, yet.

However, all the other generators that are not explicitly described as synchronous in the powsybl-dynawo model, are not necessarily non-synchronous, nor asynchronous.
I suppose these generators may be synchronous at a certain point in time, or it's a question about how simpler/complex we want to describe a network.
I don't have all the details though, I'm not well versed into the subject.
Someone from the Dynawo team would be able to elaborate more accurately about these notions.

Nonetheless, it might not be helpful to oppose synchronous/asynchronous generators. We would misinterpret the meaning of those objects, for what I understand.

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java (hunk 1 above)
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java]: Thanks for the info, I created the `AbstractGeneratorConnectedToOmegaRefModel`

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java (hunk 3 above)
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/dynamicmodels/staticref/generators/omegaref/GeneratorConnectedToOmegaRefModel.java:12]: Oops, forgot again...
Done

File: dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/dynamicmodels/GeneratorPQGroovyExtension.groovy (hunk 4 above)
Comment (dimbdr) [reply dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/dynamicmodels/GeneratorPQGroovyExtension.groovy]: Done

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/generators/AbstractGeneratorConnectedToOmegaRefModel.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * Copyright (c) 2022, RTE (http://www.rte-france.com/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package com.powsybl.dynawaltz.models.generators;
+
+public abstract class AbstractGeneratorConnectedToOmegaRefModel extends AbstractGeneratorModel implements GeneratorConnectedToOmegaRefModel {
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/models/generators/AbstractGeneratorConnectedToOmegaRefModel.java:11]: wondering if we really need this class...
- If `GeneratorPQ` is the only generator of that type, that's only to avoid duplicating `generator_omegaRefPu`
- If there could be more, ok but then we should do the same as in PR #146: the class should not be abstract and should have the lib name as constructor parameter

**Review by flo-dup — APPROVED** [review]
Just still not that happy with the `GeneratorConnectedToOmegaRefModel` name :disappointed:

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/generators/AbstractGeneratorConnectedToOmegaRefModel.java (hunk 5 above)
Comment (dimbdr) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/models/generators/AbstractGeneratorConnectedToOmegaRefModel.java]: Made the modification with the latest commits.
GeneratorConnectedToOmegaRefModel is no longer abstract, and there is additionnal info into the config file in the DSL part, that is dedicated to this object. Similar to what's done in #146
